Assets/Scripts/Managers/BattleQueue.cs

```diff
@@ -280,11 +410,426 @@ public List<ActionClass> GetList()
+        // REQUIRES: size is at least one. need a check in Dequeu ASTER7; Implicity requires that no "empty" wrappers are present i.e. -1 ones
+        public void SortWrappers()
+        {
+
+            int count = wrappers.Count;
+
+            // first sorting based on HighestSpeed
+            // need to ensure order, however, so not equals to 
+            for (int x = 0; x < count; x++)
+            {
+                int y = x;
+                while (y > 0 && wrappers[y].HighestSpeed > wrappers[y - 1].HighestSpeed)
+                {
+                    Swap(wrappers, y - 1, y);
+                    y--;
+                }
+            }
+
+
+            // now sorting based on Player Priority when the highest speed is equal player wrapper comes first and fastest player comes first e.g. 5,4 and 4,5 and 2,5 and 3,5
+            for (int x = 0; x < count; x++)
+            {
+                int y = x;
+                while (y > 0 && wrappers[y].HighestSpeed == wrappers[y - 1].HighestSpeed) // moving from right to left with the left becoming increasingly shorter.
+                {
+                    if (wrappers[y].PlayerAction == null)
+                    {
+                        y--;
+                        continue; // no swap needed 
+                    }
+                    else if (wrappers[y - 1].PlayerAction == null || wrappers[y].PlayerAction.Speed > wrappers[y - 1].PlayerAction.Speed)
+                    {
+                        Swap(wrappers, y - 1, y);
+                    }
+                    y--;
+                }
+            }
+        }
+
+
+
+
+        public List<Wrapper> GetWrappers()
+        {
+            return wrappers;
+        }
+
+    }
+
+    // Wrapper Element for WrapperArray;
+    public class Wrapper
+    {
+        public ActionClass PlayerAction { get; set; }
+        public ActionClass EnemyAction { get; set; }
+
+        public int HighestSpeed { get; set; } // used to sort the wrappers 
+                                                // -1 indicates that the wrapper is empty 
+                                                // bad encapsulation as maintenance is done outside.
+
+        // Every enemy action is transformed into a single field wrapper after all the enemy actions have been inserted;
+        // Will implement this via a field checker (roundStart) at present; should be implemented elsewhere for decorum. 
+
+        // the true corresponds to enemy action; false to player action
+        // TO REFACTOR !!! isEnemys
+        public Wrapper(ActionClass action, bool isEnemys)
+        {
+            if (isEnemys)
+            {
+                PlayerAction = null;
+                this.EnemyAction = action;
+                HighestSpeed = action.Speed;
+            }
+            else
+            {
+                this.PlayerAction = action;
+                EnemyAction = null;
+                HighestSpeed = action.Speed;
+            }
+        }
```

Comment:
consider removeing isEnemys, and just do a 

ActionClass is PlayerClass check instead